### PR TITLE
feat: Adds Prayer Pagination

### PR DESCRIPTION
**Pull request by redreceipt**
### What does this PR do, or why is it needed?

Paginates prayers!

### How do I test this PR?

- test the five types of prayers with the `prayers` (to make sure I didn't break the deprecated query)
- test the five types of prayers with the new `prayersFeed` query

## TODO

- [ ] I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))
- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] No new warnings
- [ ] Upload Screenshots/GIF(s) if applicable
- [ ] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

---

> The purpose of a PR Review is to _improve the quality of the software._

### CODE REVIEW

File: src/data/prayers/data-source.js [hunk 1]
```diff
@@ -168,19 +154,39 @@ export default class Prayer extends RockApolloDataSource {
...
     return this.sortPrayers(prayers);
   };
 
-  getFromGroups = async (groupTypeIds, personId) => {
-    const prayers = await this.request(
+  byGroups = async (groupTypeIds) => {
+    const {
+      dataSources: { Auth },
+    } = this.context;
+    const { id: personId } = await Auth.getCurrentPerson();
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply src/data/prayers/data-source.js]: fixed :)

File: src/data/prayers/data-source.js [hunk 2]
```diff
@@ -168,19 +154,39 @@ export default class Prayer extends RockApolloDataSource {
           .format()}' or ExpirationDate eq null`
       )
       .andFilter(type === 'CAMPUS' ? `CampusId eq ${primaryCampusId}` : '')
-      .get();
+      .orderBy('PrayerCount', 'asc');
+    // TODO: need to use orderBy to sort SQL side
+    // PrayerCount asc, EnteredDateTime asc
+    // need to wait for core functionality
+    // to use multiple orderBy statements
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply src/data/prayers/data-source.js]: fixed, made it in 1.3.0-beta.2

File: src/data/prayers/data-source.js [hunk 3]
```diff
@@ -168,19 +154,39 @@ export default class Prayer extends RockApolloDataSource {
...
+      dataSources: { Group },
+    } = this.context;
+    let prayersCursor;
+    if (type === 'SAVED') prayersCursor = await this.bySaved();
+    else if (type === 'GROUP')
+      prayersCursor = await this.byGroups(Group.getGroupTypeIds());
+    else prayersCursor = await this.byPrayerFeed(type);
+    const prayers = await prayersCursor.get();
     return this.sortPrayers(prayers);
   };
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply src/data/prayers/data-source.js]: I suppose so. done.

**Comment by richarddubay** [comment]
@redreceipt This is the deprecated `prayers` query looking for saved prayers. I don't have any, so this is what I get in return. I don't have any group prayers either, but that returns an empty array. I suppose this saved one should work like the group one?
Screenshot 2020-02-10 08 50 17

**Comment by richarddubay** [comment]
Actually, I get an error with the `prayerFeed` saved type version too.
Screenshot 2020-02-10 08 53 30

**Pull request merged**